proofs/src/errors.rs

```diff
@@ -1,9 +1,27 @@
+//! Error type for the `proofs` crate.
+//!
+//! This enum represents the various error conditions that can occur within the `proofs` crate.
+//! It provides a unified way to handle and propagate errors throughout the crate.
+//!
+//! The possible error variants include:
+//!
+//! - `Synthesis`: Represents an error that occurred during the synthesis process.
+//! - `Io`: Represents an I/O error.
+//! - `Serde`: Represents a serialization or deserialization error.
+//! - `Other`: Represents any other error with a custom error message.
+//! - `VerifyFailed`: Indicates that the proof verification failed.
+//! - `Parse`: Represents an error that occurred while parsing a big integer.
+//! - `WitnessCalc`: Represents an error that occurred during witness calculation (only available
+//!   when not targeting `wasm32`).
+//! - `MissingSection`: Indicates that a required section is missing.
+//! - `Bincode`: Represents a Bincode serialization or deserialization error.
 use thiserror::Error;
 
-// Wrapper for circom_witnesscalc::Error since it doesn't implement display
 #[cfg(not(target_arch = "wasm32"))]
 #[derive(Debug, Error)]
+/// Wrapper for circom_witnesscalc::Error since it doesn't implement display
 pub enum WitnessCalcError {
+  /// The error is a circom_witnesscalc::Error
   Circom(circom_witnesscalc::Error),
 }
```

Comment:
Why is this an enum? This could just be a struct since there's only one variant.

I know you didn't introduce this in this PR, i'm just saying!

Reply:
From my understanding It doesn't implement display so we had to make a wrapper type so we could implement it